pull_chart_tarball.go

```diff
@@ -11,12 +11,28 @@ import (
 
 	"github.com/giantswarm/backoff"
 	"github.com/giantswarm/microerror"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/spf13/afero"
 )
 
 // PullChartTarball downloads a tarball from the provided tarball URL,
 // returning the file path.
 func (c *Client) PullChartTarball(ctx context.Context, tarballURL string) (string, error) {
+	eventName := "pull_chart_tarball"
+
+	t := prometheus.NewTimer(histogram.WithLabelValues(eventName))
+	defer t.ObserveDuration()
+
+	chartTarballPath, err := c.pullChartTarball(ctx, tarballURL)
+	if err != nil {
+		errorGauge.WithLabelValues(eventName).Inc()
```

Comment:
Can we report chart names or at least server name (`quay.io`) as well?

Reply:
I can see the chart names or even the tarball URL as being useful. As we could alert on charts with 404s.

But all the metrics are the same and just take the event name. I don't have more time to spend on this. But maybe we can improve it as part of the Helm 3 changes?

Reply:
Sure, That would be cool.

Reply:
Thanks. I added it to the checklist in the Helm 3 ticket.